example/servers/file_server.js

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+const compression = require('compression');
+const cookieParser = require('cookie-parser');
+const cors = require('cors');
+const express = require('express');
+const fs = require("fs");
+const jwt = require('jsonwebtoken');
+const jwtMiddleware = require('express-jwt');
+const path = require('path');
+
+const AUTH_SECRET = process.env.FILE_SERVER_AUTH_SECRET || 'secret';
+const AUTH_ALGORITHM = process.env.FILE_SERVER_AUTH_ALGORITHM || 'HS256';
+const TOKEN_KEY = process.env.FILE_SERVER_TOKEN_KEY || 'token';
+
+const getDefaultToken = (req) => {
+    console.log('Get default token');
+
+    // By default, attempt to get token value in this sequence
+    // 1. headers
+    // 2. query
+    // 3. cookies
+    let token;
+    if (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer') {
+        token = req.headers.authorization.split(' ')[1];
+        console.log(`Get token from headers: ${token}`);
+    } else if (req.query && req.query.token) {
+        token = req.query.token;
+        console.log(`Get token from query: ${token}`);
+    } else if (req.cookies && req.cookies.token) {
+        token = req.cookies.token;
+        console.log(`Get token from cookies: ${token}`);
+    } else {
+        // Token not found, will return a 401 (unauthorized)
+        token = null;
+        console.log(`Token not found: ${token}`);
+    }
+    return token;
+}
+
+const getTokenBySource = (req) => {
+    const tokenSource = req.query.tokenSource;
+    console.log(`Get token by source: ${tokenSource}`);
+
+    let token;
+    switch (tokenSource) {
+    case 'headers':
+        token = req.headers.authorization.split(' ')[1];
+        console.log(`Get token from headers: ${token}`);
+        break;
+    case 'query':
+        token = req.query.token;
+        console.log(`Get token from query: ${token}`);
+        break;
+    case 'cookies':
+        token = req.cookies.token;
+        console.log(`Get token from cookies: ${token}`);
+        break;
+    default:
+        // Token not found, will return a 401 (unauthorized)
+        token = null;
+        console.log(`Token not found: ${token}`);
```

Comment:
In case of null, I'm getting error thrown in fast image server. I wonder if it's an unhandled error from the middleware or something that needs to address here.
<img width="602" alt="Screen Shot 2021-03-11 at 12 51 01 AM" src="https://user-images.githubusercontent.com/5334504/110666150-062ca000-8204-11eb-8ea8-3f6b7a4bf4c8.png">

Reply:
That's expected logging

Reply:
@saturninoabril I added an auth error handler for cleaner logging

Reply:
Thanks, LGTM!